mmd_tools/operators/morph.py

```diff
@@ -226,13 +234,23 @@ def execute(self, context):
         rig = mmd_model.Model(root)
         mmd_root = root.mmd_root
         meshObj = rig.firstMesh()
-        if meshObj is None:
-            self.report({ 'ERROR' }, "The model mesh can't be found")
-            return { 'CANCELLED' }
+
         morph = mmd_root.material_morphs[mmd_root.active_morph]
         if len(morph.data) == 0:
             return { 'FINISHED' }
         mat_data = morph.data[morph.active_material_data]
+
+        # if mmd_root.advanced_mode:
+        relMesh = rig.findMesh(mat_data.related_mesh)
+        if relMesh is not None:
+            meshObj = relMesh
+        else:
+            self.report({ 'ERROR' }, "The related mesh can't be found")
+            return { 'CANCELLED' }
+        
+        if meshObj is None:
+            self.report({ 'ERROR' }, "The model mesh can't be found")
+            return { 'CANCELLED' }        
         work_mat_name = mat_data.material+"_temp"
```

Comment:
Actually, you can refine the code from this:
>
    relMesh = rig.findMesh(mat_data.related_mesh)
    if relMesh is not None:
    meshObj = relMesh
    else:
    self.report({ 'ERROR' }, "The related mesh can't be found")
    return { 'CANCELLED' }

to this:
>
    meshObj = rig.findMesh(mat_data.related_mesh)

And other simliar places as well. :)